### PR TITLE
Add support for Clojure `tools.build` projects

### PR DESCRIPTION
**Pull request by rads**
The [`tools.build`](https://clojure.org/guides/tools_build) library is the official library for building Clojure projects from a `deps.edn` file. This PR uses the `build.clj` file for detection since technically other libraries besides `tools.build` could be built on top of `deps.edn` in the future.

Resolves https://github.com/railwayapp/nixpacks/issues/372.

### CODE REVIEW

**Comment by ndneighbor** [comment]
Will defer to @coffee-cup on what Nixpacks version this will go in. Also, it appears that CI needs to be appeased.

**Comment by rads** [comment]
@ndneighbor: Thanks for the quick review! I think I fixed the CI failures now.

**Review by ndneighbor — APPROVED** [review]

File: src/providers/clojure.rs [hunk 1]
```diff
@@ -54,6 +55,23 @@ impl Provider for ClojureProvider {
 }
 
 impl ClojureProvider {
+    fn has_lein_ring_plugin(&self, app: &App) -> bool {
+        self.is_using_lein(app)
+            && app
+                .read_file("project.clj")
+                .unwrap()
```
Comment (coffee-cup) [inline src/providers/clojure.rs:62]: Don't explicitly unwrap incase the `project.clj` file does not exist
```suggestion
                .unwrap_or_default()
```

**Review by coffee-cup — APPROVED** [review]
Looks good! Just a small comment about unwrapping

File: src/providers/clojure.rs (hunk 1 above)
Comment (rads) [reply src/providers/clojure.rs]: Thanks, I'm a total Rust newbie. 😄

**Comment by Milo123459** [comment]
Is this good to merge?

**Pull request merged**